README.md

```diff
@@ -0,0 +1,330 @@
+  ```
+- Or use convenience methods:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml").importRuntimeDependencies().resolve().withTransitivity().asFile();
+  ```
+
+- To activate profiles:
+  ```java
+  Maven.resolver().loadPomFromFile("/path/to/pom.xml", "activate-profile-1", "!disable-profile-2").importRuntimeAndTestDependencies().resolve().withTransitivity().asFile();
+  ```
+
+### Version Range Resolution
+The ShrinkWrap Resolver API allows for resolution of available versions info from a requested range. The [Maven documentation](http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html) specifies the version range syntax.
+```java
+final MavenVersionRangeResult versionRangeResult = Maven.resolver().resolveVersionRange("G:A:[1.0.0]");
+```
+
+### System properties
+
+ShrinkWrap Resolvers allows you to override any programmatic configuration via System properties.
+
+- `org.apache.maven.user-settings`
+- `org.apache.maven.global-settings`
+- `settings.security`
+- `org.apache.maven.offline`
+- `org.apache.maven.flattened-pom-path`
+- `maven.repo.local`
+- `maven.legacyLocalRepo`
+
+
+## Embedded Maven
+
+Embedded Maven allows direct invocation of Maven builds from Java code, offering functionalities such as downloading and using desired Maven binaries, an uncluttered API for simple or complex builds, additional methods for build control (e.g., ignoring failures), and easy access to build outputs and artifacts. It can be invoked using ShrinkWrap Resolver API or custom Invoker and InvocationRequest instances.
+
+### Downloading Maven Binaries
+
+Embedded Maven facilitates downloading and using specific Maven versions from Apache web pages, caching downloaded binaries, and setting up Maven home for the build.
+
+```java
+EmbeddedMaven.forProject("path/to/pom.xml").useMaven3Version(String version)
+EmbeddedMaven.forProject("path/to/pom.xml").useDistribution(URL mavenDistribution, boolean useCache)
+EmbeddedMaven.forProject("path/to/pom.xml").useInstallation(File mavenHome)
+EmbeddedMaven.forProject("path/to/pom.xml").useLocalInstallation()
+EmbeddedMaven.forProject("path/to/pom.xml").useDefaultDistribution()
+```
+
+### Additional Features
+
+Embedded Maven provides features like skipping tests, ignoring failures, logging Maven build invocation commands, and accessing built projects and their artifacts.
+
+```java
+EmbeddedMaven.forProject("path/to/pom.xml").setDebugLoggerLevel()
+EmbeddedMaven.forProject("path/to/pom.xml").setLogger(yourInvokerLogger)
+EmbeddedMaven.forProject("path/to/pom.xml").setGoals("package").build().getDefaultBuiltArchive();
+```
+
+Daemon Build
+
+For running Maven builds in the background, Embedded Maven offers daemon build functionality, enabling asynchronous execution of builds and waiting until specific output lines are matched before proceeding.
+
+```java
+EmbeddedMaven.forProject("path/to/pom.xml").setGoals("spring-boot:run").useAsDaemon().build();
+```
+
+### Experimental Features
```

Comment:
I would also add the BuiltProject somewhere, just a line or two about what it is and the "extensive" example from previous Readme